client/src/controllers/SlugController.test.js

```diff
@@ -0,0 +1,30 @@
+import { Application } from '@hotwired/stimulus';
+import { SlugController } from './SlugController';
+
+describe('SlugController', () => {
+  beforeEach(() => {
+    document.body.innerHTML = `
+      <input
+        type="text"
+        name="slug"
+        aria-describedby="panel-child-promote-child-for_search_engines-child-slug-helptext"
+        data-controller="w-slug"
+        data-action="blur->w-slug#slugify"
+        id="id_slug"
+        required=""
+        >
+      `;
+    Application.start().register('w-slug', SlugController);
+  });
+
+  it('should set slugify on the input value when focus is moved away from it', () => {
+    const slugInput = document.querySelector('#id_slug');
+    slugInput.value = 'slug testing on edit page';
+
+    const onBlur = jest.fn();
+    slugInput.addEventListener('blur', onBlur);
+    slugInput.dispatchEvent(new CustomEvent('blur'));
+
+    expect(slugInput.value).toEqual('slug-testing-on-edit-page');
+  });
```

Comment:
Would be good to add an additional test for the unicode value being non-default.